### PR TITLE
Integrate hpc coding conventions and reformat CMake files

### PR DESCRIPTION
**Pull request by pramodk**

### CODE REVIEW

**Comment by pramodk** [comment]
Locally NMODL build is failing with:

```
 cmake .. -DCMAKE_INSTALL_PREFIX=`pwd`/install -DCORENRN_ENABLE_NMODL=ON -DPYTHON_EXECUTABLE=`which python3`
make -j

[ 43%] Building CXX object external/nmodl/src/lexer/CMakeFiles/units_lexer.dir/main_units.cpp.o
[ 43%] Building CXX object external/nmodl/src/parser/CMakeFiles/c_parser.dir/main_c.cpp.o
[ 43%] Building CXX object external/nmodl/src/lexer/CMakeFiles/c_lexer.dir/main_c.cpp.o
[ 43%] Building CXX object external/nmodl/src/parser/CMakeFiles/nmodl_parser.dir/main_nmodl.cpp.o
[ 43%] Building CXX object external/nmodl/src/lexer/CMakeFiles/nmodl_lexer.dir/main_nmodl.cpp.o
/Users/kumbhar/workarena/repos/bbp/coreneuron/build/external/nmodl/src/visitors/checkparent_visitor.cpp:39:69: error: expected ')'
                "root->parent: {} is set when it should be nullptr" _format(parent_type));
                                                                    ^
/Users/kumbhar/workarena/repos/bbp/coreneuron/build/external/nmodl/src/visitors/checkparent_visitor.cpp:38:37: note: to match this '('
            throw std::runtime_error(
                                    ^
/Users/kumbhar/workarena/repos/bbp/coreneuron/build/external/nmodl/src/visitors/checkparent_visitor.cpp:48:83: error: expected ')'
            throw std::runtime_error("parent: {} and child->parent: {} missmatch" _format(
                                                                                  ^
/Users/kumbhar/workarena/repos/bbp/coreneuron/build/external/nmodl/src/visitors/checkparent_visitor.cpp:48:37: note: to match this '('
            throw std::runtime_error("parent: {} and child->parent: {} missmatch" _format(
                                    ^
2 errors generated.
make[2]: *** [external/nmodl/src/visitors/CMakeFiles/visitor_obj.dir/checkparent_visitor.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 43%] Built target symtab_obj
Scanning dependencies of target symtab
[ 43%] Linking CXX static library ../../../../lib/libsymtab.a
...
```

This issue appeared in the past but don't remember precise fix atm.

File: CMake/AddMod2cSubmodule.cmake [hunk 1]
```diff
@@ -16,24 +20,23 @@ if(NOT MOD2C_FOUND)
...
+      ${CORENEURON_PROJECT_SOURCE_DIR}/external/mod2c
+    WORKING_DIRECTORY ${CORENEURON_PROJECT_SOURCE_DIR})
 else()
   message(STATUS "Using mod2c submodule from ${MOD2C_PROJ}")
 endif()
 
 set(ExternalProjectCMakeArgs
-    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-    -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}
+    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}
```
Comment (alexsavulescu) [inline CMake/AddMod2cSubmodule.cmake:33]: Just a remark, old way was better to read

Comment (pramodk) [reply CMake/AddMod2cSubmodule.cmake:33]: 200% agree and I complained about it multiple times 😅

Newer version of cmake-format has this issue. 

Only way is to downgrade version.

cc: @ohm314

Comment (ohm314) [reply CMake/AddMod2cSubmodule.cmake:33]: I agree, guys.. but can't do much about it. Let's not fret.

**Comment by pramodk** [comment]
@ohm314 : do you remember how we solved this issue https://github.com/BlueBrain/CoreNeuron/pull/361/#issuecomment-671620241 ? we certainly encountered it on NMODL side and I don't remember what we did :) 

> don't we want to get it over with and apply coding conventions to everything at some point?

If I am not mistaken, in your branch code formatting was not applied and anyway I thought it should be separate PR. I have this in pipeline, fixing couple of things locally and will upload it for review.

**Pull request merged**